### PR TITLE
[API] Create counter for request source

### PR DESCRIPTION
**Pull request by banool**
### Description
We want some more insight into where requests to the API are coming from, this counter helps with that.

@0xmaayan this slices up the requests by source, method they're calling, and status they got. Does that sound good, or did you want the data sliced up differently?

### Test Plan
Run local testnet:
```
cargo run -p aptos -- node run-local-testnet --force-restart --assume-yes --with-faucet
```

Make a couple of requests with just curl:
```
curl http://127.0.0.1:8080/v1
curl http://127.0.0.1:8080/v1/accounts/0x1
```
Make a couple of requests with the CLI, to which we just added the User-Agent stuff.
```
aptos account list --url http://127.0.0.1:8080 --account 0x1
```

See the metrics pushed appropriately, with `request-source-client-unknown` for the former and `aptos-cli/1.8.0` for the latter:
```
curl -s http://127.0.0.1:9101/metrics | grep -i request_source
```
```
# HELP aptos_api_request_source_client API requests grouped by source (e.g. which SDK, unknown, etc), operation_id, and status
# TYPE aptos_api_request_source_client counter
aptos_api_request_source_client{operation_id="estimate_gas_price",request_source_client="request-source-client-unknown",status="200"} 1
aptos_api_request_source_client{operation_id="get_account",request_source_client="request-source-client-unknown",status="200"} 1
aptos_api_request_source_client{operation_id="get_account",request_source_client="request-source-client-unknown",status="404"} 1
aptos_api_request_source_client{operation_id="get_account_resources",request_source_client="aptos-cli/1.0.8",status="200"} 1
aptos_api_request_source_client{operation_id="get_ledger_info",request_source_client="request-source-client-unknown",status="200"} 2
aptos_api_request_source_client{operation_id="get_transaction_by_hash",request_source_client="request-source-client-unknown",status="200"} 13
aptos_api_request_source_client{operation_id="submit_transaction",request_source_client="request-source-client-unknown",status="202"} 1
```

### CODE REVIEW

**Review by 0xmaayan — APPROVED** [review]
LGTM! thanks!

File: api/types/src/headers.rs [hunk 1]
```diff
@@ -17,3 +17,5 @@ pub const X_APTOS_OLDEST_BLOCK_HEIGHT: &str = "X-Aptos-Oldest-Block-Height";
 pub const X_APTOS_LEDGER_TIMESTAMP: &str = "X-Aptos-Ledger-TimestampUsec";
 /// Cursor used for pagination.
 pub const X_APTOS_CURSOR: &str = "X-Aptos-Cursor";
+/// Included in requests from clients we provide to identify the request source.
+pub const X_APTOS_SOURCE_CLIENT: &str = "X-Aptos-Source-Client";
```
Comment (gregnazario) [inline api/types/src/headers.rs:21]: How is this different than the User-Agent header?  Should get that for free automatically from SDKs and others, and is more common across sources.

Comment (banool) [reply api/types/src/headers.rs:21]: We spoke offline, we'll use User-Agent. See https://github.com/aptos-labs/aptos-core/pull/7061 for more discussion on this topic.

**Review by gregnazario — APPROVED** [review]

File: api/src/log.rs [hunk 2]
```diff
@@ -2,16 +2,22 @@
...
     prelude::{sample, SampleRate},
     warn, Schema,
 };
+use once_cell::sync::Lazy;
 use poem::{http::header, Endpoint, Request, Response, Result};
 use poem_openapi::OperationId;
+use regex::Regex;
 use std::time::Duration;
 
+const REQUEST_SOURCE_CLIENT_UNKNOWN: &str = "request-source-client-unknown";
```
Comment (gregnazario) [inline api/src/log.rs:17]: We could probably just shorten this to just `unknown`, it's contextual so we can save on some bytes.
